cmd/zdb/zdb.c

```diff
@@ -6608,7 +6608,11 @@ zdb_read_block(char *thing, spa_t *spa)
 			}
 			error = zio_wait(czio);
 			if (error == 0 || error == ECKSUM) {
-				zio_checksum_compute(czio, ck, pabd, lsize);
+				zio_t ck_zio = { 0 };
```

Comment:
I'd suggest using `ck_zio = zio_root(spa, NULL, NULL, 0)` here and destroy it with `zio_wait()` below.  This way everything in the zio is initialized so if at some point `zio_checksum_compute()` is modified to depend on other fields it won't introduce a problem.

Reply:
Thanks I took your suggestion.